tests/e2e/utils.py

```diff
@@ -0,0 +1,436 @@
+    0.025418603792786598,
+    -0.0019248499302193522,
+    -0.04490964487195015,
+    0.05094093829393387,
+    -0.007762531749904156,
+    0.0022712810896337032,
+    -0.048982247710227966,
+    0.04316263645887375,
+    0.03385719656944275,
+    -0.02204400859773159,
+    -0.008660282008349895,
+    -0.0358533076941967,
+    0.11571227014064789,
+    -0.11273631453514099,
+    0.06849292665719986,
+    -0.03595784679055214,
+    0.04161537066102028,
+    -0.013967745006084442,
+    -0.004196549765765667,
+    0.04396875202655792,
+    0.07947150617837906,
+    -0.038232240825891495,
+    -0.004747415892779827,
+    -0.012230468913912773,
+    0.039619915187358856,
+    -0.11451079696416855,
+    0.0012758959783241153,
+    -0.007560120429843664,
+    0.015182306058704853,
+    -0.039508406072854996,
+    -0.06489536166191101,
+    -0.003909661900252104,
+    0.0037100922781974077,
+    0.010909723117947578,
+    -0.07211877405643463,
+    0.024324269965291023,
+    0.04425453022122383,
+    0.02042572572827339,
+    -0.07636857777833939,
+    -0.03965488821268082,
+    -0.04338192194700241,
+    -0.10347922146320343,
+    -0.04393737390637398,
+    -0.01368421409279108,
+    -1.568378671379378e-08,
+    0.10027588903903961,
+    -0.043567851185798645,
+    0.050843071192502975,
+    -0.08256983011960983,
+    0.02588729001581669,
+    0.056297384202480316,
+    -0.010702059604227543,
+    0.08518678694963455,
+    0.02700129710137844,
+    0.07154033333063126,
+    -0.05953506380319595,
+    0.0778793916106224,
+    0.036633629351854324,
+    0.08145243674516678,
+    0.07252412289381027,
+    -0.002973137190565467,
+    -0.013765356503427029,
+    -0.046995650976896286,
+    -0.0247399490326643,
+    -0.027804242447018623,
+    -0.025088943541049957,
+    -0.01294120866805315,
+    -0.03852837160229683,
+    0.11392410844564438,
+    0.030737342312932014,
+    -0.065057672560215,
+    0.06248151883482933,
+    -0.01869017630815506,
+    0.02368609979748726,
+    -0.0654701292514801,
+    0.058893367648124695,
+    0.06793640553951263,
+    0.008447104133665562,
+    0.015255128964781761,
+    -0.04546090587973595,
+    -0.04216332733631134,
+    0.02343529835343361,
+    -0.07064592838287354,
+    0.01851677894592285,
+    -0.006000120658427477,
+    -0.025140153244137764,
+    0.003474486293271184,
+    -0.01600506715476513,
+    0.0020763161592185497,
+    0.010104725137352943,
+    0.0021877381950616837,
+    0.07691455632448196,
+    0.008766746148467064,
+    -0.012105572037398815,
+    -0.047985345125198364,
+    -0.012077542021870613,
+    -0.02539162151515484,
+    0.01326005533337593,
+    -0.05451769009232521,
+    -0.08339792490005493,
+    0.052714310586452484,
+    -0.011956708505749702,
+    -0.01412263885140419,
+    -0.05299930274486542,
+    0.0125426622107625,
+    0.12013623863458633,
+    0.09160523116588593,
+    0.12442110478878021,
+    -0.03318800404667854,
+]
+
+
+def populate_neo4j(neo4j_driver, neo4j_objs):
```

Comment:
Sorry for keep nagging about this one, but since it's tied to the `jeopardy_tiny_....json` and the structure of the data object, I think we want to encapsulate more here (creating the `neo4j_objs` + writing to the db).

Reply:
Will probably be helpful for the RAG testing as well. Though it can be done in the PR dedicated to this feature.

Reply:
No worries. I've done a bit more refactoring and moved the `build_data_objects` function to `tests/e2e/utils.py`. Pinecone and Weaviate each require a slightly different structure for their vectors when writing so I've had to include a `q_vector_fmt` argument so the `question_objs` are returned in the correct format. Let me know if you think there's any more encapsulation to be done here.